nlp/2021_11_25_gpt3mix/nlp_gpt3mix.ipynb

```diff
@@ -0,0 +1,2575 @@
+      },
+      "source": [
+        "!!pip install -qq transformers datasets tokenizers openai requests"
+      ],
+      "execution_count": null,
+      "outputs": [
+        {
+          "output_type": "execute_result",
+          "data": {
+            "text/plain": [
+              "[]"
+            ]
+          },
+          "metadata": {},
+          "execution_count": 1
+        }
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "Ir6zD__FU84n"
+      },
+      "source": [
+        "### Imports"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "hM-09Chsj7wI"
+      },
+      "source": [
+        "import re\n",
+        "import json\n",
+        "import torch\n",
+        "import random\n",
+        "import requests\n",
+        "import numpy as np\n",
+        "import pandas as pd \n",
+        "import plotly.express as px\n",
+        "import plotly.graph_objects as go\n",
+        "\n",
+        "from plotly.subplots import make_subplots\n",
+        "from datasets import load_dataset, concatenate_datasets, load_from_disk, load_metric, Dataset\n",
+        "from transformers import pipeline, AutoTokenizer, AutoModelForSequenceClassification, TrainingArguments, Trainer, TrainerCallback"
+      ],
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "gFQ6LEgjeRWb"
+      },
+      "source": [
+        "### Download dataset\n",
+        "We'll use [Emotion](https://huggingface.co/datasets/emotion) that is a dataset of English Twitter messages labeled as one of the six basic emotions: anger, fear, joy, love, sadness and surprise. To make our task a bit easier, we will use only three of them, namely:\n",
+        "- joy 😂\n",
+        "- anger 😠\n",
+        "- surprise 😯"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "1ltsurmvaaGd",
+        "colab": {
+          "base_uri": "https://localhost:8080/",
+          "height": 159,
+          "referenced_widgets": [
+            "b39348e38bfa4ff6a7266b023f0c58a8",
+            "85d21cfd0acc41fcb3e9aca459820c7a",
+            "f11d9c96a18b4ef0828de6b55ebf7fb9",
+            "5a5c40173f704c26bf4942d7217841d1",
+            "5ff00fca536c49e28bf6e8dce5a259a3",
+            "84e282b8ef4646eab5df9f41c39a911d",
+            "c87bdfc339be47999ec49770d6ee67ec",
+            "8134d4170465444c8b9965eb27bb3f28",
+            "ac4431d9a3c9481dafe08955ea2b5217",
+            "564be917b26d46889ca710db83a61a20",
+            "76417a8169554b359b889b3cbda96ebb"
+          ]
+        },
+        "outputId": "fdb8b35d-b298-4a9b-9410-1fd6db991bdb"
+      },
+      "source": [
+        "# load the dataset and filter on samples that have a token count less than 30 to use only short tweets\n",
+        "max_input_len = 30\n",
+        "tokenizer = AutoTokenizer.from_pretrained(\"distilbert-base-cased\")\n",
```

Comment:
Nitpick, but actually, if you look at the dataset, you'll see that the data seems to be lowercase already. So casing information won't add a lot of performance to the model. So in this case `distilbert-base-uncased` might make more sense